e2e_playwright/host_config_test.py

```diff
@@ -52,3 +59,34 @@ def test_disable_fullscreen(
     assert_snapshot(
         dataframe_toolbar, name="host_config-dataframe_disabled_fullscreen_mode"
     )
+
+
+def test_block_error_dialogs(page: Page, app_port: int):
+    """Test that error dialogs are blocked and sent to host when set via host-config."""
+    # Need to be more specific about the route to allow for successful redirect
+    page.route(
+        f"http://localhost:{app_port}/_stcore/host-config",
+        handle_route_hostconfig_disable_fullscreen_and_error_dialogs,
+    )
+
+    # Initial load of page
+    page.goto(f"http://localhost:{app_port}")
+    wait_for_app_loaded(page)
+
+    # Capture console messages
+    messages = []
+    page.on("console", lambda msg: messages.append(msg))
+
+    # Navigate to a non-existent page to trigger page not found error
+    page.goto(f"http://localhost:{app_port}/nonexistent_page")
+    page.wait_for_timeout(1000)
+
+    # Console includes 2 404 errors (health & host-config) + 1 error from page not found (last message)
+    last_message = messages.pop()
+    assert (
```

Comment:
nit: This looks a bit risky regarding flakiness. Maybe it's a bit better to use `wait,_until` helper from conftest checking if the message is in messages via the callback (instead of the wait_for_timeout & assert)

Reply:
Makes sense, thanks!